### PR TITLE
R4DS 2e blog post + Learn update

### PR DESCRIPTION
**Pull request by mine-cetinkaya-rundel**
This PR does two things:

- Adds a new blog post announcing R4DS 2e:
  - Some of the text is original and some is verbatim from the [preface to the 2nd edition](https://r4ds.hadley.nz/preface-2e.html), I want to make sure that's ok.
- Updates learn page:
  - Links to 2e of the book -- Do I have the correct link with the referral id? It was a shortlink before.
  - Updates existing other learning resources, mostly links that went stale over time.
  - Adds a new section on teaching materials -- I thought this might be a place for this, but I'm happy to leave it out if it doesn't feel like the right fit.

### CODE REVIEW

File: content/blog/r4ds-2e/index.Rmd [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+TODO:
+* [ ] Look over / edit the post's title in the yaml
+* [ ] Edit (or delete) the description; note this appears in the Twitter card
+* [ ] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [ ] Find photo & update yaml metadata
+* [ ] Create `thumbnail-sq.jpg`; height and width should be equal
+* [ ] Create `thumbnail-wd.jpg`; width should be >5x height
+* [ ] `hugodown::use_tidy_thumbnails()`
+* [ ] Add intro sentence, e.g. the standard tagline for the package
+* [ ] `usethis::use_tidy_thanks()`
```
Comment (hadley) [inline content/blog/r4ds-2e/index.Rmd:31]: I think you've done most of these?

File: content/blog/r4ds-2e/index.Rmd [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+output: hugodown::hugo_document
+
+slug: r4ds-2e
+title: R for Data Science, 2nd edition
+date: 2023-07-11
+author: Mine Çetinkaya-Rundel
+description: >
+    The second edition of R for Data Science is out,
+    and it's a major reworking of the first edition.
+photo:
```
Comment (hadley) [inline content/blog/r4ds-2e/index.Rmd:11]: This needs an update. Would be nice to use a photo of the actual book?

File: content/blog/r4ds-2e/index.Rmd [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+* [ ] `hugodown::use_tidy_thumbnails()`
+* [ ] Add intro sentence, e.g. the standard tagline for the package
+* [ ] `usethis::use_tidy_thanks()`
+-->
+```
+We're thrilled to announce the publication of the 2nd edition of [R for Data Science](https://r4ds.hadley.nz/).
+
+The second edition is a major reworking of the first edition, removing material we no longer think is useful, adding material we wish we included in the first edition, and generally updating the text and code to reflect changes in best practices.
+
+You can read the book online for free at [https://r4ds.hadley.nz](https://r4ds.hadley.nz/){.uri}, or [buy a physical copy](https://www.amazon.com/Data-Science-Transform-Visualize-Model/dp/1492097403/ref=sr_1_1?crid=19VLZ2HBNTMG2).
```
Comment (hadley) [inline content/blog/r4ds-2e/index.Rmd:38]:
```suggestion
You can read the book online for free at [https://r4ds.hadley.nz](https://r4ds.hadley.nz/){.uri}, or [buy a physical copy](https://www.amazon.com/dp/1492097403?&tag=hadlwick-20).
```

File: content/blog/r4ds-2e/index.Rmd [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+Finally, the [\"Communicate\"](https://r4ds.hadley.nz/communicate.html) part remains, but has been thoroughly updated to feature [Quarto](https://quarto.org/) instead of R Markdown. This edition of the book has been written in Quarto, and it\'s clearly the tool of the future.
+
+## What's gone?
+
+The first edition of the book featured a part on modeling, which has now been removed. We never had enough room to fully do modelling justice, and there are now much better resources available. We generally recommend using the [tidymodels](https://www.tidymodels.org/) packages and reading [Tidy Modeling with R](https://www.tmwr.org/) by Max Kuhn and Julia Silge.
+
+## Acknowledgements
+
+This book isn\'t just the product of Hadley, Mine, and Garrett but is the result of many conversations (in person and online) that we\'ve had with many people in the R community. Huge thanks to [all contributors](And, as always, issues and pull requests welcome on the book repository.) for the conversations, issues, and pull requests. And, as always, issues and pull requests welcome on the [book repository](https://github.com/hadley/r4ds/).
```
Comment (hadley) [inline content/blog/r4ds-2e/index.Rmd:62]: Bad link formatting for "All contributors"?

**Review by hadley — APPROVED** [review]

File: content/blog/r4ds-2e/index.Rmd (hunk 2 above)
Comment (mine-cetinkaya-rundel) [reply content/blog/r4ds-2e/index.Rmd:11]: Yup, I've made images with actual photo of the book, neglected to update the photo URL. Should I link to https://www.oreilly.com/library/view/r-for-data/9781492097396/ with "O'Reilly" as the text or https://r4ds.hadley.nz/ with "R for Data Science" as the text, any preference?

Comment (hadley) [reply content/blog/r4ds-2e/index.Rmd:11]: I meant like take a photo of it sitting on your desk 😄

**Pull request merged**